src/harness/reference_models/interference/interference.py

```diff
@@ -0,0 +1,648 @@
+    # constraint frequency range 3550-3660MHz
+    if ProtectedEntityType.ESC == _DISTANCE_PER_PROTECTION_TYPE[constraint.entity_type]:
+      if grant.cbsd_category == 'A' and constraint.high_frequency > ESC_CAT_A_HIGH_FREQ_HZ:
+        freq_check = False
+
+    # Append the grants information if it is inside the neighborhood of
+    # protection constraint
+    if freq_check:
+      grants_inside.append(grant)
+
+  return grants_inside
+
+
+def getGrantObjectsFromFAD(sas_uut_fad_object, sas_th_fad_objects):
+  """Extracts CBSD grant objects from FAD object
+
+  Args:
+    sas_uut_fad_object: FAD object from SAS UUT
+    sas_th_fad_object: a list of FAD objects from SAS Test Harness
+  Returns:
+    grant_objects: a list of CBSD grants dictionary containing registrationRequest
+    and grants
+  """
+  # List of CBSD grant tuples extracted from FAD record
+  grant_objects = []
+
+  # Creating list of cbsds
+  cbsd_list_uut = []
+  cbsd_list_th = []
+
+  for cbsds in sas_uut_fad_object.getCbsdRecords():
+    cbsd_list_uut.append(cbsds)
+
+  grant_objects_uut = getAllGrantInformationFromCbsdDataDump(cbsd_list_uut, True)
+
+  for fad in sas_th_fad_objects:
+    for cbsds in fad.getCbsdRecords():
+      cbsd_list_th.append(cbsds)
+      grant_objects_test_harness = getAllGrantInformationFromCbsdDataDump(
+                                     cbsd_list_th, False)
+
+  grant_objects = grant_objects_uut + grant_objects_test_harness
+
+  return grant_objects 
+
+def getAllGrantInformationFromCbsdDataDump(cbsd_data_records, is_managing_sas=True):
+  """Extracts list of CbsdGrantInformation namedtuple
+
+  Routine to extract CbsdGrantInformation tuple from CBSD data records from 
+  FAD objects
+
+  Args:
+    cbsd_data_records: A list CbsdData object retrieved from FAD records.
+    is_managing_sas: flag indicating cbsd data record is from managing SAS or 
+                     peer SAS
+                     True - Managing SAS, False - Peer SAS
+  Returns:
+    grant_objects: a list of grants, each one being a namedtuple of type 
+                   CBSDGrantInformation, of all CBSDs from SAS UUT FAD and 
+                   SAS Test Harness FAD
+  """
+
+  grant_objects = []
+
+  # Loop over each CBSD grant
+  for cbsd_data_record in cbsd_data_records:
+    registration = cbsd_data_record.get('registrationRequest')
+    grants = cbsd_data_record.get('grants')
+
+    # Check CBSD location
+    lat_cbsd = registration.get('installationParam', {}).get('latitude')
+    lon_cbsd = registration.get('installationParam', {}).get('longitude')
+    height_cbsd = registration.get('installationParam', {}).get('height')
+    height_type_cbsd = registration.get('installationParam', {}).get('heightType')
+    if height_type_cbsd == 'AMSL':
+      altitude_cbsd = terrainDriver.GetTerrainElevation(lat_cbsd, lon_cbsd)
+      height_cbsd = height_cbsd - altitude_cbsd
+
+    # Sanity check on CBSD antenna height
+    if height_cbsd < 1 or height_cbsd > 1000:
```

Comment:
This does not belong to this piece of code. 
It is currently done in propag models, but there has been agreement among SAS admins that all heights
below 1m (or 1.5m TBD) will be automtically reinitialized to 1m. This is to copde with the fact of errors in reporting
AGL height or inconsistencies in AMSL reported height vs terrain data.
So best is to remove this from here, and let the prop model handle this

Reply:
Removed   height check